### PR TITLE
Reload confirmation page after error and added webhook

### PR DESCRIPTION
**Pull request by abdyfranco**

### CODE REVIEW

File: views/default/settings.pdt [hunk 1]
```diff
@@ -16,6 +16,14 @@
 		</li>
 	</ul>
 
+    <div class="title_row">
+        <h3><?php $this->_('StripePayments.webhook'); ?></h3>
+    </div>
+    <div class="pad">
+        <p><?php $this->_('StripePayments.webhook_note'); ?></p>
+        <code><?php echo Configure::get('Blesta.gw_callback_url') . Configure::get('Blesta.company_id') . '/stripe_payments/'; ?></code>
```
Comment (JReissmueller) [inline views/default/settings.pdt:24]: Currently, the callback only handles nonmerchant gateways

Comment (abdyfranco) [reply views/default/settings.pdt:24]: I created a PR for the core, adding support for merchant gateways.

**Review by JReissmueller — CHANGES_REQUESTED** [review]
The processStoredAch method evaluates pending transactions as approved. With the addition of this callback system that is no longer necessary.  Instead it should mark the transaction as pending

File: views/default/settings.pdt (hunk 1 above)
Comment (JReissmueller) [reply views/default/settings.pdt:24]: Got it.  The link generated by this uses /callback/gw instead of /calback/mgw

File: stripe_payments.php [hunk 2]
```diff
@@ -1433,4 +1455,85 @@ private function global($key, $value = null)
...
+        // Get event payload
+        $payload = @file_get_contents('php://input');
+        if (!empty($payload)) {
+            $payload = json_decode($payload);
+        } else {
+            $payload = (object) [];
+        }
+
+        // Validate only payment intent events
+        if ($payload->data->object->object !== 'payment_intent') {
```
Comment (JReissmueller) [inline stripe_payments.php:1489]: This should also support charges for the sake of validating ACH callbacks

**Review by JReissmueller — CHANGES_REQUESTED** [review]

File: stripe_payments.php [hunk 3]
```diff
@@ -1433,4 +1455,85 @@ private function global($key, $value = null)
...
+            return false;
+        }
+
+        // Get event status
+        $status = 'error';
+        $stripe_status = $payload->data->object->charges->data[0]->status ?? $payload->data->object->status ?? 'failed';
+        if (isset($stripe_status)) {
+            switch ($stripe_status) {
+                case 'requires_capture':
+                case 'requires_payment_method':
```
Comment (JReissmueller) [inline stripe_payments.php:1513]: Charges may have the status 'pending'

**Review by JReissmueller — CHANGES_REQUESTED** [review]

**Review by JReissmueller — APPROVED** [review]

**Pull request merged**